gobblin-runtime/src/main/java/gobblin/runtime/mapreduce/MRJobLauncher.java

```diff
@@ -198,6 +199,17 @@ protected void runJob(String jobName, Properties jobProps, JobState jobState, Li
     // Turn off speculative execution
     this.job.setSpeculativeExecution(false);
 
+    // Delete any staging directories that already exist
```

Comment:
Can you add this to AbstractJobLauncher.runWorkUnits(), which already has flattened WorkUnits?